scripts/js/commands/api/syncDevDocs.ts

```diff
@@ -67,12 +67,25 @@ async function getArtifactUrl(
   repoName: string,
   workflowId: string,
 ): Promise<string> {
+  // Note that we do not filter for successful workflow runs, which we'd do by putting
+  // `and .conclusion=="success"` in the `select()`. This is because Runtime frequently fails
+  // its integration tests. The docs jobs are deterministic though so should be fine.
   const runProc =
-    await $`gh api repos/qiskit/${repoName}/actions/workflows/${workflowId}/runs --jq '.workflow_runs[] | select(.head_branch=="main" and .conclusion=="success") | .id'`;
+    await $`gh api repos/qiskit/${repoName}/actions/workflows/${workflowId}/runs --jq '.workflow_runs[] | select(.head_branch=="main") | .id'`;
   const runId = runProc.stdout.split("\n")[0];
-  const url =
+  if (!runId) {
+    throw new Error(`Failed to get run ID for ${repoName}`);
+  }
+
+  const urlProc =
     await $`gh api repos/qiskit/${repoName}/actions/runs/${runId}/artifacts --jq '.artifacts[0].archive_download_url'`;
-  return url.stdout.trim();
+  const url = urlProc.stdout.trim();
+  if (!url) {
+    throw new Error(
+      `Failed to get artifact URL for https://github.com/Qiskit/${repoName}/actions/runs/${runId}`,
+    );
```

Comment:
Should we throw or just `console.error`? I assume throwing will kill regeneration for all packages, even if only one fails

Reply:
I think throw. If something is wrong, I want to know about it so that we can investigate. No one checks logs for cron jobs, so we'd risk an API not being updated for a long time.

We can always manually generate locally or loosen this if we find a good reason. But I'd rather start defensive.